workflow/controller/operator.go

```diff
@@ -2387,6 +2404,30 @@ func (woc *wfOperationCtx) initializeNode(nodeName string, nodeType wfv1.NodeTyp
 	return &node
 }
 
+// Update a node status with cache status
+func (woc *wfOperationCtx) updateAsCacheNode(node *wfv1.NodeStatus, memStat *wfv1.MemoizationStatus) *wfv1.NodeStatus {
+	node.MemoizationStatus = memStat
+
+	woc.wf.Status.Nodes[node.ID] = *node
```

Comment:
This should use `woc.wf.Status.Nodes.Set` instead. It was introduced super recently in https://github.com/argoproj/argo-workflows/pull/11451

Reply:
More safety. thanks.